src/Datadog.Trace/FrameworkDescription.cs

```diff
@@ -0,0 +1,156 @@
+using System;
+using System.Reflection;
+using System.Text.RegularExpressions;
+
+namespace Datadog.Trace
+{
+    internal class FrameworkDescription
+    {
+        private static FrameworkDescription _instance;
+
+        private FrameworkDescription(
+            string name,
+            string productVersion,
+            string fileVersion,
+            Version assemblyVersion,
+            Version environmentVersion,
+            string osPlatform,
+            string osArchitecture,
+            string processArchitecture)
+        {
+            Name = name;
+            ProductVersion = productVersion;
+            FileVersion = fileVersion;
+            AssemblyVersion = assemblyVersion;
+            EnvironmentVersion = environmentVersion;
+            OSPlatform = osPlatform;
+            OSArchitecture = osArchitecture;
+            ProcessArchitecture = processArchitecture;
+        }
+
+        public static FrameworkDescription Instance
+        {
+            get
+            {
+                if (_instance == null)
+                {
+                    _instance = Create();
+                }
+
+                return _instance;
+            }
+        }
+
+        public string Name { get; }
+
+        public string ProductVersion { get; }
+
+        public string FileVersion { get; }
+
+        public Version AssemblyVersion { get; }
+
+        public Version EnvironmentVersion { get; }
+
+        public string OSPlatform { get; }
+
+        public string OSArchitecture { get; }
+
+        public string ProcessArchitecture { get; }
+
+        public static string GetOSPlatform()
+        {
+            switch (Environment.OSVersion.Platform)
+            {
+                case PlatformID.Win32NT:
+                    return "win";
+                case PlatformID.Unix:
```

Comment:
Since this is in the Trace library and is not tied to the profiler, we should also make sure to include MacOS as an OS Platform that we report, since that is supported on .NET Core 1.0+

Reply:
Good point. Same reply as [here](https://github.com/DataDog/dd-trace-dotnet/pull/519#discussion_r329846778).

Reply:
Good point. Same reply as [here](https://github.com/DataDog/dd-trace-dotnet/pull/519#discussion_r329846778).